### ISSUE TITLE
Analyzer load failures due to MS.CA version need a designated ID

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#20316]
Currently if an analyzer references a newer version of the compiler assemblies than the build is using, warning CS8032 is reported. This warning is indistinguishable from warnings produced from other kinds of load failures, which is leading to confusion and multiple customer bug reports.

Users need to be able to install an analyzer in a project under the following conditions:

* The analyzer is built against version X of the compiler assemblies (Microsoft.CodeAnalysis.\*)
* The project compiles under version X (or newer) of the compiler, and the analyzers run
* The project compiles under version Y<X of the compiler, and the analyzers do not run
* No warning is reported related to the analyzer assemblies under either of the two builds above
* If an analyzer fails to load due to a dependency load which is *not* related to Microsoft.CodeAnalysis version, then that warning *is* reported for both of the above builds

Unfortunately, due to the overly-broad use of CS8032, the latter two items are indistinguishable from each other, forcing the warning to be disable altogether and losing its value.

**Comment by mavasani** [comment]
Yes, this would be helpful. We can also reconsider https://github.com/dotnet/roslyn/issues/2683 to improve other messages related to analyzer load failures

**Comment by CyrusNajmabadi** [comment]
Closing out as speculative.
